syscall.go

```diff
@@ -115,10 +117,18 @@ func unmountShareDir() error {
 	return os.RemoveAll(containerMountDest)
 }
 
-func waitForBlockDevice(deviceName string) error {
+func waitForBlockDevice(deviceName string, isSCSIAddr bool) error {
 	devicePath := filepath.Join(devPath, deviceName)
 
-	if _, err := os.Stat(devicePath); err == nil {
+	if isSCSIAddr {
+		// Here in "0:0", the first number is the SCSI host number, while the second number is always 0.
+		scsiPrefix := "/sys/class/scsi_disk/0:0"
+
+		path := scsiPrefix + deviceName + "/device/block"
+		if _, err := os.Stat(path); err == nil {
+			return nil
+		}
+	} else if _, err := os.Stat(devicePath); err == nil {
```

Comment:
I think it would be clearer if you moved the `devicePath` variable down to here and removed the `else if`:

```
if isSCSIAddr {
    :
}

devicePath := filepath.Join(devPath, deviceName)

if _, err := os.Stat(devicePath); err == nil {
    return nil
}


```